src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -129,10 +130,13 @@ public DatasetService(
   public String createDataset(
       DatasetRequestModel datasetRequest, AuthenticatedUserRequest userReq) {
     String description = "Create dataset " + datasetRequest.getName();
-    loggingMetrics.set(
-        BardEventProperties.BILLING_PROFILE_ID_FIELD_NAME, datasetRequest.getDefaultProfileId());
+    UUID defaultProfileId = datasetRequest.getDefaultProfileId();
+    loggingMetrics.set(BardEventProperties.BILLING_PROFILE_ID_FIELD_NAME, defaultProfileId);
     return jobService
         .newJob(description, DatasetCreateFlight.class, datasetRequest, userReq)
+        .addParameter(JobMapKeys.IAM_RESOURCE_TYPE.getKeyName(), IamResourceType.SPEND_PROFILE)
```

Comment:
I wonder: would it make sense to introduce an object that contains these three values rather than send in three scalars every sine since they all get used together? Something like "ExposeTo" or "RequiredAcls" (not sure about the naming) that would look like `{"resourceType":"...", "resourceId": "...", "action": "..."}` but that could theoretically also lead to arrays of these.